### PR TITLE
Refactor homepage and database repository

### PR DESCRIPTION
**Pull request by Hila00**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -24,4 +21,4 @@ class MyApp extends StatelessWidget {
       routes: MovieRouter.routes(),
     );
   }
-}
+}
```
Comment (BANDL011) [inline lib/main.dart:24]: Add end of file

File: lib/features/movies/data/repository/implementation/database_repository_impl.dart [hunk 2]
```diff
@@ -19,60 +21,70 @@ abstract class IDatabaseRepository {
...
-  ) async {
+      MovieModel movie,
+      Endpoints endpoint,
+      ) async {
+    MoviesDatabase dbInstance = await moviesDatabase;
+    MoviesDao moviesDao = dbInstance.moviesDao;
     MovieModel? existingMovie = await getMovieById(movie);
     if (existingMovie != null) {
-      if (!existingMovie.category.contains(endpoint.endpointName),) {
+      if (!existingMovie.category.contains(endpoint.endpointName)) {
```
Comment (LuciaMBandeo) [inline lib/features/movies/data/repository/implementation/database_repository_impl.dart:39]: Missing trailling comma

File: lib/main.dart [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
 import 'routing/movie_router.dart';
 
 void main() async {
   WidgetsFlutterBinding.ensureInitialized();
 
   runApp(
-    const ProviderScope(
-      child: MyApp(),
-    ),
+    const ProviderScope(child: MyApp()),
```
Comment (LuciaMBandeo) [inline lib/main.dart]: Missing trailling comma

File: lib/features/movies/data/repository/implementation/database_repository_impl.dart (hunk 2 above)
Comment (LuciaMBandeo) [inline lib/features/movies/data/repository/implementation/database_repository_impl.dart:39]: Missing trailling comma

File: lib/main.dart (hunk 1 above)
Comment (LuciaMBandeo) [inline lib/main.dart]: Fix end of file

**Review by LuciaMBandeo — APPROVED** [review]

**Pull request merged**